core/src/lightclient/mod.rs

```diff
@@ -354,4 +367,130 @@ impl StarknetClient for StarknetClientImpl {
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Declare(DeclareTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Deploy(DeployTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::DeployAccount(
+                        DeployAccountTransactionReceipt {
+                            transaction_hash,
+                            actual_fee,
+                            status,
+                            block_hash,
+                            block_number,
+                            ..
+                        },
+                    ) => {
+                        res_receipt.transaction_hash =
+                            Some(PrimitiveH256::from_slice(&transaction_hash.to_bytes_be()));
+                        res_receipt.gas_used = Some(felt_to_u256(actual_fee));
+                        res_receipt.status_code = match status {
+                            StarknetTransactionStatus::Pending => Some(U64::from(0)),
+                            StarknetTransactionStatus::AcceptedOnL1 => Some(U64::from(1)),
+                            StarknetTransactionStatus::AcceptedOnL2 => Some(U64::from(1)),
+                            StarknetTransactionStatus::Rejected => Some(U64::from(0)),
+                        };
+                        res_receipt.block_hash =
+                            Some(PrimitiveH256::from_slice(&block_hash.to_bytes_be()));
+                        res_receipt.block_number = Some(felt_to_u256(block_number.into()));
+                    }
+                };
+            }
+            MaybePendingTransactionReceipt::PendingReceipt(_) => {
+                return Ok(None);
+            }
+        };
+
+        let starknet_tx = self.client.get_transaction_by_hash(hash_felt).await?;
+        match starknet_tx.clone() {
+            StarknetTransaction::Invoke(invoke_tx) => {
+                match invoke_tx {
+                    InvokeTransaction::V0(v0) => {
+                        res_receipt.contract_address =
+                            Some(starknet_address_to_ethereum_address(v0.contract_address));
+                    }
+                    InvokeTransaction::V1(_) => {}
+                };
+            }
+            StarknetTransaction::L1Handler(l1_handler_tx) => {
```

Comment:
It's not clear to me whether or not we can interact with L1 - L2 messaging.

I would suggest adding a TODO comment above that refers to Kakarot not handling L1-L2 messaging